lib/WTSI/DNAP/Warehouse/Schema/Result/EseqRun.pm

```diff
@@ -0,0 +1,250 @@
+
+package WTSI::DNAP::Warehouse::Schema::Result::EseqRun;
+
+# Created by DBIx::Class::Schema::Loader
+# DO NOT MODIFY THE FIRST PART OF THIS FILE
+
+=head1 NAME
+
+WTSI::DNAP::Warehouse::Schema::Result::EseqRun
+
+=head1 DESCRIPTION
+
+Information about a run performed on an Element Biosciences instrument derived from instrument output
+
+=cut
+
+use strict;
+use warnings;
+
+use Moose;
+use MooseX::NonMoose;
+use MooseX::MarkAsMethods autoclean => 1;
+extends 'DBIx::Class::Core';
+
+=head1 COMPONENTS LOADED
+
+=over 4
+
+=item * L<DBIx::Class::InflateColumn::DateTime>
+
+=back
+
+=cut
+
+__PACKAGE__->load_components('InflateColumn::DateTime');
+
+=head1 TABLE: C<eseq_run>
+
+=cut
+
+__PACKAGE__->table('eseq_run');
+
+=head1 ACCESSORS
+
+=head2 id_eseq_run_tmp
+
+  data_type: 'bigint'
+  extra: {unsigned => 1}
+  is_auto_increment: 1
+  is_nullable: 0
+
+Internal to this database id, value can change
+
+=head2 folder_name
+
+  data_type: 'varchar'
+  is_nullable: 0
+  size: 200
+
+Run folder name created by the Element Biosciences instrument
+
+=head2 run_name
+
+  data_type: 'varchar'
+  is_nullable: 0
+  size: 100
+
+Run name as recorded in RunParameters.json file
+
+=head2 flowcell_id
+
+  data_type: 'varchar'
+  is_nullable: 0
+  size: 100
+
+Flowcell ID as recorded in RunParameters.json file
+
+=head2 date_started
+
+  data_type: 'datetime'
+  datetime_undef_if_invalid: 1
+  is_nullable: 1
+
+Run date as recorded in RunParameters.json file
+
+=head2 date_completed
+
+  data_type: 'datetime'
+  datetime_undef_if_invalid: 1
+  is_nullable: 1
+
+Run completion date from a timestamp of RunUploaded.json file
+
+=head2 run_parameters
+
+  data_type: 'json'
+  is_nullable: 0
+
+The content of RunParameters.json file in the run folder
+
+=head2 run_stats
+
+  data_type: 'json'
+  is_nullable: 1
+
+The content of AvitiRunStats.json file in the run folder
+
+=head2 outcome
```

Comment:
Good news! We've just had a walk-up run that failed at the end, but the data is fine and it deplexes fine. I'm assuming we record the "truth" here, and such oddities get communicated by hand to invested parties.

Reply:
Eventually, when a QC system is in place, this kind of good data is QC-ed and it will be present in iRODS, which will overwrite the failure recorded by the instrument.